### PR TITLE
Write parameters from ParameterMap

### LINKED ISSUE #148: Handle writing parameters to file in ParameterMap
## Describe the missing feature
Convenient writing of current parameters to file.

## Describe the solution you'd like
Currently, the optimizer writes parameters to file when updated, both into a log of the optimization run and a separate 'best point' file. Let's move this functionality to the ParameterMap, to become a counterpart to the `load_values` method.

## Describe alternatives you've considered
Leave as is.

### PR DESCRIPTION
**Pull request by nwittler**
## What
Introduce a method to HJSON dump current parameter values.

## Why
Closes #148 

## How
The code to store the current best point of the optimization was moved from the `c3/optimizer/optimizer.py` super class to `c3/parametermap.py`. Optimizers now use the method.

## Example use
In a jupyter or script use `pmap.store_values("save_point.c3log")` to create a file that contains:
```
{
  opt_map:
  [
    [
      cr90-d1-flattop-amp
    ]
    [
      cr90-d1-flattop-freq_offset
    ]
    [
      cr90-d1-flattop-xy_angle
    ]
  ]
  units:
  [
    V
    Hz
    pi
  ]
  optim_status:
  {
    params:
    [
      1
      0.5
      0.123
    ]
  }
}
```

The counterpart is `pmap.load_values("save_point.c3log")`

## Checklist
Please include and complete the following checklist. Your Pull Request is (in most cases) not ready for review until the following have been completed. You can create a draft PR while you are still completing the checklist. Check the [Contribution Guidelines](https://github.com/q-optimize/c3/blob/dev/CONTRIBUTING.md) for more details. You can mark an item as complete with the `- [x]` prefix

- [x] Tests - Added unit tests for new code, regression tests for bugs and updated the integration tests if required
- [x] Formatting & Linting - `black` and `flake8` have been used to ensure styling guidelines are met
- [x] Type Annotations - All new code has been type annotated in the function signatures using type hints
- [x] Docstrings - Docstrings have been provided for functions in the `numpydoc` style
- [x] Documentation - The tutorial style documentation has been updated to explain changes & new features
- [x] Notebooks - Example notebooks have been updated to incorporate changes and new features
- [ ] Changelog - A short note on this PR has been added to the Upcoming Release section

### CODE REVIEW

**Comment by lazyoracle** [comment]
This seems to be a tensorflow keras version conflict issue because I assume the tests are actually fine (they pass the code coverage CI).

Ref: [stackoverflow](https://stackoverflow.com/questions/68970841/another-metric-with-the-same-name-already-exists)

**Comment by nwittler** [comment]
> This seems to be a tensorflow keras version conflict issue because I assume the tests are actually fine (they pass the code coverage CI).
> 
> Ref: [stackoverflow](https://stackoverflow.com/questions/68970841/another-metric-with-the-same-name-already-exists)

It's also breaking the tests for #140 that I just merged. The test can fine a while ago, before the merge.

**Comment by lazyoracle** [comment]
Yes, I will try to go in and check if this is something restricted certain versions of python (unless somebody else goes ahead and does that).

**Review by lazyoracle — CHANGES_REQUESTED** [review]
Can you include a note or code snippet somewhere in the docs or example notebooks etc to show an example way of using this feature to write the current best parameters?

**Comment by nwittler** [comment]
> Can you include a note or code snippet somewhere in the docs or example notebooks etc to show an example way of using this feature to write the current best parameters?

I added a tutorial on the Parametermap.

**Review by lazyoracle — APPROVED** [review]
Ready to merge
